frontend/src/App.js

```
@@ -652,159 +652,188 @@
+            <div className="chart-container d-flex justify-content-center align-items-center" style={{minHeight: '300px'}}>
+                <div className="text-center text-muted">
+                    <i className="bi bi-bar-chart fs-1 mb-2"></i>
+                    <p>No category data available</p>
+                </div>
+            </div>
+        );
+    }
 
-    // Create/update chart when data changes and Chart.js is loaded
+    // Create chart effect
     useEffect(() => {
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

Hooks should not be called after an early return.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)